kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -275,6 +268,6 @@ private void initOperationContext(Config configuration) {
     if (CustomResource.class.isAssignableFrom(apiTypeClass)) {
       return CustomResourceDefinitionContext.fromCustomResourceType(apiTypeClass).getPlural();
     }
-    return getPluralFromKind(HasMetadata.getKind(apiTypeClass));
+    return Pluralize.toPlural(HasMetadata.getKind(apiTypeClass)).toLowerCase();
```

Comment:
That should really be: `CustomResource.getPlural(apiTypeClass)`. The problem, though, is that this requires `apiTypeClass` to extend `CustomResource`, which is why I tend to think that the logic for these methods should maybe be moved to `HasMetadata`. Either that, or make these informers methods take a `CustomResource` instance?
The problem with the logic above is that potentially the class has been annotated with `@Singular` which means that the singular version might not be equal to kind. The class might also have been annotated with `@Plural` in which case, that value should be used and not some other computation…

Reply:
Yes, I think you're right. We should update `sharedIndexInformerForCustomResource` to use `CustomResource` instead. I'll update this.